src/server/render.ts

```diff
@@ -145,6 +149,44 @@ export async function render<Data>(
     ? defaultCsp()
     : undefined;
   const headComponents: ComponentChildren[] = [];
+  if (csp) {
+    // Clear the csp
+    const newCsp = defaultCsp();
+    csp.directives = newCsp.directives;
+    csp.reportOnly = newCsp.reportOnly;
+  }
+  // Clear the head components
+  headComponents.splice(0, headComponents.length);
+
+  // Setup the interesting VNode types
+  ISLANDS.splice(0, ISLANDS.length, ...opts.islands);
+
+  // Clear the encountered vnodes
+  ENCOUNTERED_ISLANDS.clear();
+
+  // Clear the island props
+  ISLAND_PROPS = [];
+
+  let component = opts.route.component;
+  if (
+    typeof component === "function" &&
+    component.constructor.name === "AsyncFunction"
+  ) {
```

Comment:
```suggestion
  if (component.constructor.name === "AsyncFunction") {
```
This should suffice, no?